packages/typespec-ts/src/modelUtils.ts

```diff
@@ -453,6 +453,8 @@ function getSchemaForModel(
     true /** shouldGuard */
   );
 
+  modelSchema.typeName = modelSchema.name;
```

Comment:
This is the only change. not sure why Health Insight fail to generate the propertyType name if that property is a Model, I tried to repro it in the unit-test, still unable to repo.

Reply:
All models in property failed or just some specific one?

Reply:
all.

Reply:
Approved but still prefer to find the root cause to repro in UT.